Node is currently this only EventTarget subclass actually requiring CanMakeCheckedPtr

cdumez opened the pull request:
#### b9a502d60c06019a202360058c8435f683f3e2f3
<pre>
Node is currently this only EventTarget subclass actually requiring CanMakeCheckedPtr
<a href="https://bugs.webkit.org/show_bug.cgi?id=266256">https://bugs.webkit.org/show_bug.cgi?id=266256</a>

Reviewed by Darin Adler.

Node is currently this only EventTarget subclass actually requiring CanMakeCheckedPtr.
We can thus have Node subclass CanMakeCheckedPtr directly to decrease the size of
every other EventTarget subclass.

* Source/WebCore/dom/EventTarget.cpp:
* Source/WebCore/dom/EventTarget.h:
* Source/WebCore/dom/MessagePort.cpp:
(WebCore::MessagePort::disentanglePorts):
* Source/WebCore/dom/Node.cpp:
* Source/WebCore/dom/Node.h:

Canonical link: <a href="https://commits.webkit.org/271905@main">https://commits.webkit.org/271905@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/917616b7c118f996e4b4a086c6cdb4dc07fc5058

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30011 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8674 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31324 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32513 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27128 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30671 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10864 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5925 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/27170 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30305 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7275 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25596 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6211 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/6368 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/26689 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33850 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27370 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27121 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/32546 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/6308 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/4481 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30342 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8052 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7108 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7057 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6835 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/917616b7c118f996e4b4a086c6cdb4dc07fc5058. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/21654

darinadler reviewed (approved):
A little sneaky that your patch makes this true with a code change, but the title doesn’t say that. But I wouldn’t waste time trying to fix that.

I guess this fits with our theme of using `WeakPtr` more since it’s not a lot less efficient than `CheckedPtr`?

cdumez commented:
> A little sneaky that your patch makes this true with a code change, but the title doesn’t say that. But I wouldn’t waste time trying to fix that.
> 
> I guess this fits with our theme of using `WeakPtr` more since it’s not a lot less efficient than `CheckedPtr`?

Yes, EventTargets are ref counted so we can hold a Ref / RefPtr for stack objects (no need for CheckedRef / CheckedRef). For non-stack objects, WeakPtr / WeakRef would be preferred.

cdumez committed:
Node is currently this only EventTarget subclass actually requiring CanMakeCheckedPtr
https://bugs.webkit.org/show_bug.cgi?id=266256

Reviewed by Darin Adler.

Node is currently this only EventTarget subclass actually requiring CanMakeCheckedPtr.
We can thus have Node subclass CanMakeCheckedPtr directly to decrease the size of
every other EventTarget subclass.

* Source/WebCore/dom/EventTarget.cpp:
* Source/WebCore/dom/EventTarget.h:
* Source/WebCore/dom/MessagePort.cpp:
(WebCore::MessagePort::disentanglePorts):
* Source/WebCore/dom/Node.cpp:
* Source/WebCore/dom/Node.h:

Canonical link: https://commits.webkit.org/271905@main

webkit-commit-queue commented:
Committed 271905@main (b9a502d60c06): <https://commits.webkit.org/271905@main>

Reviewed commits have been landed. Closing PR #21654 and removing active labels.
